### ISSUE TITLE
Incorrect Training Code

### ISSUE THREAD

**Issue by dumbgoos** [protdb/PowerNovo#2]
Hi,

I hope you’re doing well. I encountered an issue while running the training process using the trainer file in your repository. Specifically, on line 32, the code calls `self.model.peptide_decoder_forward()`, but I couldn’t find a definition for this function in the codebase. Should this perhaps be `self.model.decode()`?

After changing it to `decode()`, I ran into further errors. The tensors `decoder_logit` and `tokens_ids` appear to be 2D, which causes an error with the line:
```python
decoder_logit = decoder_logit.permute(0, 2, 1)
```
When I removed that line, the subsequent loss calculation:
```python
loss = self.loss(decoder_logit, tokens_ids)
```
also resulted in an error.

Could you please provide the correct version of the trainer file or offer some guidance on how to resolve these issues? Your help is greatly appreciated.

Thank you for your time and assistance.

Best regards

**Comment by Denis21800** [comment]
Hello, I uploaded the latest version of the file train.py

**Comment by dumbgoos** [comment]
> Hello, I uploaded the latest version of the file train.py

Thank you very much for your reply, the code is ready to run successfully.
